src/ExpressionEvaluator/Core/Source/ResultProvider/Helpers/MemberAndDeclarationInfo.cs

```diff
@@ -246,8 +247,71 @@ private sealed class MemberNameComparer : IComparer<MemberAndDeclarationInfo>
             public int Compare(MemberAndDeclarationInfo x, MemberAndDeclarationInfo y)
             {
                 var comp = string.Compare(x.Name, y.Name, StringComparison.Ordinal);
-                return comp != 0 ? comp : (y._inheritanceLevel - x._inheritanceLevel);
+                if (comp == 0)
+                {
+                    return y._inheritanceLevel - x._inheritanceLevel;
+                }
+
+                // Data members come first
+                var xHoldsData = HoldsData(x);
+                var yHoldsData = HoldsData(y);
+                if (xHoldsData && yHoldsData)
+                {
+                    return comp;
+                }
+                if (xHoldsData)
+                {
+                    return -1;
+                }
+                if (yHoldsData)
+                {
+                    return 1;
+                }
+
+                return comp;
+
+                static bool HoldsData(MemberAndDeclarationInfo member)
+                {
+                    if (member.MemberType == MemberTypes.Field)
+                    {
+                        return true;
+                    }
+
+                    if (member._member is PropertyInfo propertyInfo)
+                    {
+                        return IsCompilerGenerated(propertyInfo.GetGetMethod(nonPublic: true)) || IsCompilerGenerated(propertyInfo.GetSetMethod(nonPublic: true));
```

Comment:
>IsCompilerGenerated(propertyInfo.GetGetMethod(nonPublic: true)) || IsCompilerGenerated(propertyInfo.GetSetMethod(nonPublic: true)) [](start = 31, length = 130)

Are auto-properties ordered before other properties? As a user, I think I'd expect all properties ordered by name. Anything else might look like a bug.

Reply:
We have 3 people and 4 opinions on what the behavior should be. :-/
I will revert to my original design (ie. just tweaking the lexical ordering, without relying on semantic of data vs. non-data member).

Reply:
To clarify, the three possible behaviors that we discussed so far (and were implemented in turn...):
1) change the lexical order of members (`_` and `x_` come before the rest)
2) fields come first
3) data members come first
4) do nothing

Those options are sorted in my order of preference. Reviewers, what are your thoughts?

Reply:
I suspect options 2 and 3 will produce ordering that will be difficult to understand and cause users to search for members. 

---
In reply to: [358351799](https://github.com/dotnet/roslyn/pull/40210#discussion_r358351799) [](ancestors = 358351799)

Reply:
I don't think there is much difference between [2] and [3] in potential confusion, since fields may be CamelCase named. So if we do any of these two let's go with [3].
That said, without visual cue it might be confusing to do [3]. So you might want to table it until the debugger changes UI (file an issue to track) and do just [1].

Reply:
I'm okay to go back to [1] as well. I'll update the PR tomorrow morning.
Please chime in this afternoon if you prefer otherwise. Thanks